core/lib/src/config/secret_key.rs

```diff
@@ -178,6 +193,80 @@ impl SecretKey {
     {
         ser.serialize_bytes(&[0; 32][..])
     }
+
+    /// Encrypts the given data.
+    /// Generates a random nonce for each encryption to ensure uniqueness.
+    /// Returns the Vec<u8> of the concatenated nonce and ciphertext.
+    ///
+    /// # Example
+    /// ```rust
+    /// use rocket::config::SecretKey;
+    ///
+    /// let plaintext = "I like turtles".as_bytes();
+    /// let secret_key = SecretKey::generate().unwrap();
+    ///
+    /// let encrypted = secret_key.encrypt(&plaintext).unwrap();
+    /// let decrypted = secret_key.decrypt(&encrypted).unwrap();
+    ///
+    /// assert_eq!(decrypted, plaintext);
+    /// ```
+    pub fn encrypt<T: AsRef<[u8]>>(&self, value: T) -> Result<Vec<u8>, Error> {
```

Comment:
A return type of `Vec<u8>` is not particularly useful. It would be nice to return something that encapsulates the `Vec<u8>` with convenient methods, like [`blake3::Hash`](https://docs.rs/blake3/latest/blake3/struct.Hash.html). For example, something lIke:

```rust
struct Cipher(Vec<u8>);

impl Cipher {
    fn from_bytes(&[u8]) -> Result<Self>;
    fn from_vec(Vec<u8>) -> Result<Self>;
    fn from_hex(&str) -> Result<Self>;
    fn from_base64(&str) -> Result<Self>;

    fn as_bytes(&self) -> &[u8];
    fn into_vec(self) -> Vec<u8>;
    fn to_hex(&self) -> String;
    fn to_base64(&self) -> String;
}
```